contentcuration/contentcuration/frontend/shared/feedbackApiUtils.js

```diff
@@ -0,0 +1,148 @@
+    for (const key in this) {
+      if (
+        Object.prototype.hasOwnProperty.call(this, key) &&
+        typeof this[key] !== 'function' &&
+        key !== 'URL'
+      ) {
+        dataObject[key] = this[key];
+      }
+    }
+    return dataObject;
+  }
+
+  // Return URL associated with the ObjectType
+  getUrl() {
+    if (this.defaultURL === null || this.URL === undefined) {
+      throw new Error('URL is not defined for the FeedBack Object.');
+    }
+    return this.URL;
+  }
+}
+
+/**
+ * Initializes a new BaseFeedbackEvent object.
+ *
+ * @param {Object} params - Parameters for initializing the object.
+ * @param {Object} params.user - The user associated with the feedback event.
+ * @param {string} params.target_channel_id - The ID of the target channel for the feedback event.
+ * @param {Object} baseFeedbackParams - Additional parameters inherited from the base feedbackclass.
+ */
+// eslint-disable-next-line no-unused-vars
+class BaseFeedbackEvent extends BaseFeedback {
+  constructor({ user, target_channel_id, ...baseFeedbackParams }) {
+    super(baseFeedbackParams);
+    this.user = user;
+    this.target_channel_id = target_channel_id;
+  }
+}
+
+/**
+ * Initializes a new BaseFeedbackInteractionEvent object.
+ * @param {Object} params - Parameters for initializing the interaction event.
+ * @param {string} params.feedback_type - The type of feedback for the interaction event.
+ * @param {string} params.feedback_reason - The reason associated with the feedback.
+ * @param {Object} baseFeedbackParams - Additional parameters inherited from the base feedbackclass.
+ */
+class BaseFeedbackInteractionEvent extends BaseFeedback {
+  constructor({ feedback_type, feedback_reason, ...baseFeedbackParams }) {
+    super(baseFeedbackParams);
+    this.feedback_type = feedback_type;
+    this.feedback_reason = feedback_reason;
+  }
+}
+
+/**
+ * Initializes a new BaseFlagFeedback object.
+ *
+ * @param {Object} params - Parameters for initializing the flag feedback.
+ * @param {string} params.target_topic_id - The ID of the target topic associated with
+ * the flag feedback.
+ * @param {Object} baseFeedbackParams - Additional parameters inherited from the
+ * base interaction event class.
+ */
+class BaseFlagFeedback extends BaseFeedbackInteractionEvent {
+  constructor({ target_topic_id, ...baseFeedbackParams }) {
+    super({ ...baseFeedbackParams });
+    this.target_topic_id = target_topic_id;
+  }
+}
+
+/**
+ * Initializes a new FlagFeedbackEvent object.
+ *
+ *
+ * @param {Object} params - Parameters for initializing the flag feedback event.
+ * @param {string} params.target_topic_id - Id of the topic where the flagged content is located.
+ * @param {Object} baseFeedbackParams - Additional parameters inherited from the
+ * base flag feedback class.
+ */
+export class FlagFeedbackEvent extends BaseFlagFeedback {
+  constructor({ target_topic_id, ...baseFeedbackParams }) {
+    super({ target_topic_id, ...baseFeedbackParams });
```

Comment:
@vkWeb mentioned that, it seems like target-topic_id can be ommited here. We can access the parent_id field of the contentNode to access the target_topic_id.